src/main/java/com/zaxxer/hikari/pool/PoolElf.java

```diff
@@ -183,9 +183,10 @@ void setupConnection(final Connection connection, final long connectionTimeout)
       connection.setAutoCommit(isAutoCommit);
       connection.setReadOnly(isReadOnly);
 
-      final int defaultLevel = connection.getTransactionIsolation();
-      transactionIsolation = (transactionIsolation < 0 ? defaultLevel : transactionIsolation);
-      if (transactionIsolation != defaultLevel) {
```

Comment:
*Only if* the user specified level was different than the default, we set the connection isolation level.  If the user did not specifiy, an isolation, this is always false, and we don't set anything.